bin/src/main.rs

```diff
@@ -239,6 +241,28 @@ async fn _main() {
         }
     };
 
+    // TODO Leader Election
+    let metric_stats_source = match config.log.log_metric_server_stats {
```

Comment:
I think this should be moved up to line ~220 or so, inside the match block before we release the lease.

This would also let us reuse the k8s_client created there, just need to move this block up and change the return from that block to `(k8s_event_stream, metric_stats_source)` then adjust the let binding up at line 162 to unpack it